crates/proof-of-sql/src/sql/proof/provable_result_column.rs

```diff
@@ -7,64 +6,60 @@ use crate::{
 /// Interface for serializing an intermediate result column
 pub trait ProvableResultColumn {
     /// The number of bytes of the serialized result column
-    fn num_bytes(&self, selection: &Indexes) -> usize;
+    fn num_bytes(&self, length: u64) -> usize;
 
     /// Serialize the result column
-    fn write(&self, out: &mut [u8], selection: &Indexes) -> usize;
+    fn write(&self, out: &mut [u8], length: u64) -> usize;
 }
 
 impl<'a, T: ProvableResultElement<'a>> ProvableResultColumn for &[T] {
-    fn num_bytes(&self, selection: &Indexes) -> usize {
-        let mut res = 0;
-        for i in selection.iter() {
-            res += self[i as usize].required_bytes();
-        }
-        res
+    fn num_bytes(&self, length: u64) -> usize {
+        (0..length as usize).map(|i| self[i].required_bytes()).sum()
```

Comment:
Comment: I have a feeling that `length` may not even be needed. We could test this easily by adding:
```suggestion
    fn num_bytes(&self, length: u64) -> usize {
        assert_eq!(self.len(), length)
        self.iter().map(|e| e.required_bytes()).sum()
```